firstvoices/backend/views/custom_order_recalculate_views.py

```
@@ -0,0 +1,81 @@
+            return Response(
+                {"message": "No recalculation results found for this site."}, status=404
+            )
+
+        # If there is a task_id, check the status of the task and add it to the response
+        if self.task_id:
+            async_result = AsyncResult(self.task_id)
+            preview_info = {"preview_current_task_status": async_result.status}
+        else:
+            preview_info = {"preview_current_task_status": "Not started."}
+
+        # If there is a result, add it to the response
+        if result:
+            preview_info["latest_recalculation_date"] = result.date
+            preview_info["latest_recalculation_result"] = result.result
+        else:
+            preview_info["latest_recalculation_date"] = None
+            preview_info["latest_recalculation_result"] = {}
+
+        return Response(preview_info, status=200)
+
+    def post(self, request, site_slug: str):
+        # Check Superadmin status
+        self.check_superadmin_status(request)
+
+        # Check that the site exists
+        try:
+            site = Site.objects.get(slug=site_slug)
+        except Site.DoesNotExist:
+            return Response({"message": "Site not found"}, status=404)
+
+        # Call the recalculation preview task
+        try:
+            recalculation_results = recalculate_custom_order_preview.apply_async(
+                (site_slug,)
+            )
+            self.task_id = recalculation_results.task_id
+
+            CustomOrderRecalculationPreviewResult.objects.create(
+                site=site, result=recalculation_results.get(timeout=360)
+            )
+
+            return Response(
+                {"message": "Successfully saved recalculation results"}, status=201
+            )
+
+        except recalculate_custom_order_preview.OperationalError:
+            return Response(
+                {"message": "An error occurred while dispatching remote task"},
+                status=503,
```

I think a 500 makes more sense here. 503 usually means the server is too busy or your request got throttled

---

Also, please raise an exception instead of manually constructing an error response. (If we change our default error response format in future, we don't want to have to separately edit this one.) It's probably worth making a custom APIException subclass for this. E.g., https://www.django-rest-framework.org/api-guide/exceptions/#apiexception

---

I've changed this to raise the built in drf server error (500) but if you'd prefer a custom APIException I can make one